### PR TITLE
Feat/Novedades/Crear reducer de novedades

### PR DESCRIPTION
**Pull request by TelmaKn**
_**##SUMMARY**_

Using redux-toolkit this PR adds a reducer to manage the status of the news and moves the existing logic of requests to the API to asynchronous thunks within the reducer.
In the beginning the work was done only with the reducer and the service, making some arrangements in the service so that everything works well. When everything worked in network and redux, when bringing the state logic to the visual components, several arrangements had to be made in the different components to be able to close the flow of data. For this, the component's edition path had to be added to be able to edit the different news by accessing from the edition button of the Backoffice.
I must add that being close to the date of the demo, I tried that what has to do with the data flow of the news works as well as possible and in this it is possible that some other existing problem that was not completely within it has been solved Of the scope perhaps, but it had to do with the handling of News data.

_**##CHANGES ADDED**_

-Adds the News Reducer file with the functions of Create, Edit, DeleteById, GetAll, GetById and CreateorUpdate. It also adds the representation of the reducer in the store.
-Fix problems with the service for optimal operation of the Network and the corresponding connection with the API.
-Fix problems in the Editing and News Creation Form file such as writing that works in reverse or setting the url of the file. For the latter, it was necessary to change the Input File component since it was not possible to achieve the required operation with which it was and the url did not allow the information to travel. Changes were also necessary in the setting of values ​​of the inputs so that everything arrives as it was due to redux and the API and is rendered in the form in case of being a news edition.
-Adds the functionalities required for state management to the Backoffice table component and at this point adds a path to which the edit button present in this file can be directed.
-Solves fixes in the card components so that the redux information reaches them correctly when rendering.
-Finally refactor the code to make it as clean and understandable as possible.

**_I apologize from now on if some formatting left any changes confusing due to quotes or spaces, I did my best to make it not happen, but there were many commits._**

_**##USERSTORY**_

https://alkemy-labs.atlassian.net/jira/software/c/projects/OT91/boards/145?modal=detail&selectedIssue=OT91-107

_**##EVIDENCE**_

https://www.loom.com/share/8431bd70df2040f4b06157795b5f9c0d

### CODE REVIEW

File: src/Components/News/Detail/NewsDetailLayout.js [hunk 1]
```diff
@@ -14,29 +16,24 @@ const NewsImage = lazy(
 );
 
 const NewsDetailLayout = () => {
-  const [newData, setNewData] = useState({});
   const [newsDescription, setNewsDescription] = useState("");
-  const [loading, setIsLoading] = useState(true);
+  const [loading, setIsLoading] = useState(false);
```
Comment (saralavanchy) [inline src/Components/News/Detail/NewsDetailLayout.js:20]: why you are changing this to false? with this change it will never show the loader

**Review by saralavanchy — CHANGES_REQUESTED** [review]

File: src/Components/News/Detail/NewsDetailLayout.js (hunk 1 above)
Comment (TelmaKn) [reply src/Components/News/Detail/NewsDetailLayout.js:20]: Oh no! I did it in some test just to check a few things and I forgot like this! sorry! I fix it!

Comment (TelmaKn) [reply src/Components/News/Detail/NewsDetailLayout.js:20]: Sara, in the final PR the loading was fine, in true. Apparently I returned it to true in some other commit.

**Review by saralavanchy — DISMISSED** [review]

**Comment by saralavanchy** [comment]
Excelent PR, perfect description, congrats!!!

**Review by saralavanchy — APPROVED** [review]

**Pull request merged**